pkg/procmatch/readme.md

```diff
@@ -0,0 +1,63 @@
+# procmatch
+
+Procmatch is a library that provides a way to extract datadog-integrations from command lines by using a catalog.
+
+## Getting Started
+
+### Installing
+
+To install the package: `go get -u github.com/DataDog/datadog-agent/pkg/procmatch`
+
+Usage:
+
+```golang
+package main
+
+import (
+	"fmt"
+
+	"github.com/DataDog/datadog-agent/pkg/procmatch"
+)
+
+func main() {
+    matcher, err := procmatch.NewDefault()
+    if err != nil {
+        // Handle the error
+    }
+
+    cmd := "java org.elasticsearch.bootstrap.elasticsearch"
+    fmt.Printf("Integration: %s\n", matcher.Match(cmd).Name)
+    // Outputs:
+    // Integration: elastic
+}
+```
+
+The default catalog is available [here](/default_catalog.go)
+
+## Adding a new integration's signature
+
+To add a new integration signature, you will first have to add it inside its `manifest.json` file in the [datadog integration-core repo](https://github.com/DataDog/integrations-core) under the `process_signatures` field.
+
+After that create a PR on the datadog integration-core repo to commit your change.
+
+Then you can regenerate the default catalog to get the new signature here (see section below to regenerate the catalog)
+
+For testing purposes you can also modify the generated file `/default_catalog.go`.
+
+## Generate a new catalog
+
+The catalog is generated [here](/gen/generate_catalog.go), you will first need to clone the [datadog integration-core repo](https://github.com/DataDog/integrations-core).
+
+Then you can run `INTEGRATIONS_CORE_DIR=<path_to_integration_repo> go generate` inside _procmatch_ directory.
```

Comment:
nit: like the instructions on `Running the tests` below, the generation command could simply be `INTEGRATIONS_CORE_DIR=<path_to_integration_repo> go generate ./...` (or `INTEGRATIONS_CORE_DIR=<path_to_integration_repo> go generate ./pkg/procmatch/` if we assume we're at the repo root